src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -775,6 +773,9 @@ bool ClusterResourceScheduler::AllocateTaskResourceInstances(
         }
       }
     } else {
+      // Allocation failed. Restore node's local resources by freeing the resources
+      // of the failed allocation.
+      FreeTaskResourceInstances(task_allocation);
```

Comment:
@wuisawesome I think it's a bug here, but not sure, could you confirm it?

Reply:
@wuisawesome  in case you missed this one

Reply:
@wuisawesome please verify if this makes sense. @iycheng is this related to this PR? Can we make a separate PR for the fix?

Reply:
I think it's just a fix for this one but feel like worth separating for it. I'm not sure whether it's easy to add a unit test for this as well. Lack of bandwidth for that if it's too difficult. I'm ok to roll it back and create an issue if you prefer.

Reply:
So it seems like the scenario you are thinking is the allocation only contains the subset of custom resources right? 

e.g., 

Task: custom_1: 1.0, custom_2: 1.0

Node: custom_1: 1.0

and this won't free custom_1.

I think this should be easy to unit test? If we add this in this PR, I think we should add unit test at least

Reply:
Ok, then let me roll it back for now and create a p1 ticket for this.

Reply:
@rkooo567 I think this code path will only be reached in this case:

- placement group created
- schedule with that pg (before running this function)
- placement group returned
- fail to allocate the resource for placement group

The case you mentioned won't trigger this error, since it won't even be scheduled to this node.